js/base/functions/crypto.js

```diff
@@ -26,11 +27,36 @@ const hmac = (request, secret, hash = 'sha256', digest = 'hex') => {
 
 /*  .............................................   */
 
-const jwt = function JSON_web_token (request, secret, alg = 'HS256', hash = 'sha256') {
+/**
+ * @return {string}
+ */
+const jwt = function JSON_web_token (request, secret, alg = 'HS256') {
+    const algos = {
+            'HS256': 'sha256',
+            'HS384': 'sha384',
+            'HS512': 'sha512',
+            'RS256': 'pkcs1-sha256',
+            'RS512': 'pkcs1-sha512',
+        };
     const encodedHeader = urlencodeBase64 (stringToBase64 (JSON.stringify ({ 'alg': alg, 'typ': 'JWT' })))
-        , encodedData = urlencodeBase64 (stringToBase64 (JSON.stringify (request)))
-        , token = [ encodedHeader, encodedData ].join ('.')
-        , signature = urlencodeBase64 (utf16ToBase64 (hmac (token, secret, hash, 'utf16')))
+    const encodedData = urlencodeBase64 (stringToBase64 (JSON.stringify (request)))
+    const token = [ encodedHeader, encodedData ].join ('.')
+    if (!(alg in algos)) {
+        throw new ExchangeError (alg + ' is not a supported jwt algorithm.')
+    }
+    const algoType = alg.slice (0, 2);
+    const algorithm = algos[alg]
+    let signature = undefined
+    if (algoType === 'HS') {
+        signature = urlencodeBase64 (utf16ToBase64 (hmac (token, secret, algorithm, 'utf16')))
+    } else if (algoType === 'RS') {
+        let key = new NodeRSA (secret, {
+            'environment': 'browser',
```

Comment:
according to the [docs](https://github.com/rzcoder/node-rsa) this should mean it doesn't use `crypto`

Reply:
"alledgedly"

Reply:
@frosty00 ah, ok... however, if that import isn't referenced anywhere in utils.js – we can still strip it away to avoid side-effects )

Reply:
I think it does use crypto in the sign method looking at it now ):

Reply:
make we should find the source for crypto and vendor it

Reply:
But not it utils.js at least ) Do you mean that it won't work with browsers because of that?

Reply:
@frosty00 what about this one: http://www-cs-students.stanford.edu/~tjw/jsbn/ ?

Reply:
Basically, as far as I understand, RS256 = RSA signing + SHA256 hashing, and we have SHA256 hashing, so we just need RSA signing, right? Or not?

Reply:
That code literally use `crypto` its part of this (and most) js rsa implementations

Reply:
This is the code I was looking at this morning https://github.com/kjur/jsrsasign except it is really hard to debug

Reply:
> what order he put them together into the minified script >?>

https://github.com/kjur/jsrsasign/blob/master/Makefile

I think it's defined here, check it out ↑

Reply:
... basically, we need just the rsasign.js, and its deps...